File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/items/media/files/nowplaying/NowPlayingToggledVisibilityControls.java
package com.lasthopesoftware.bluewater.client.library.items.media.files.nowplaying;

import android.view.View;
import android.widget.LinearLayout;
import android.widget.RatingBar;

import com.lasthopesoftware.bluewater.shared.view.LazyViewFinder;

/**
 * Created by david on 10/16/15.
 */
public class NowPlayingToggledVisibilityControls {
    private final LazyViewFinder<LinearLayout> playerControlsLinearLayout;
    private final LazyViewFinder<LinearLayout> menuControlsLinearLayout;
    private final LazyViewFinder<RatingBar> ratingBar;

    private boolean isVisible = true;

    public NowPlayingToggledVisibilityControls(LazyViewFinder<LinearLayout> playerControlsLinearLayout, LazyViewFinder<LinearLayout> menuControlsLinearLayout, LazyViewFinder<RatingBar> ratingBar) {
        this.playerControlsLinearLayout = playerControlsLinearLayout;
        this.menuControlsLinearLayout = menuControlsLinearLayout;
        this.ratingBar = ratingBar;
    }

    public void toggleVisibility(boolean isVisible) {
        this.isVisible = isVisible;

        final int normalVisibility = isVisible ? View.VISIBLE : View.INVISIBLE;

        playerControlsLinearLayout.findView().setVisibility(normalVisibility);
        ratingBar.findView().setVisibility(normalVisibility);
        menuControlsLinearLayout.findView().setVisibility(isVisible ? View.VISIBLE : View.GONE);
    }

    public boolean isVisible() {
        return isVisible;
    }
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/playback/view/nowplaying/activity/NowPlayingToggledVisibilityControls.kt
package com.lasthopesoftware.bluewater.client.playback.view.nowplaying.activity

import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import com.lasthopesoftware.bluewater.shared.android.view.LazyViewFinder

internal class NowPlayingToggledVisibilityControls(
	private val nowPlayingListHandle: LazyViewFinder<ImageView>,
    private val playerControlsLinearLayout: LazyViewFinder<LinearLayout>,
    private val menuControlsLinearLayout: LazyViewFinder<LinearLayout>,
    private val ratingBarLinearLayout: LazyViewFinder<LinearLayout>
) {
    var isVisible = true
        private set

    fun toggleVisibility(isVisible: Boolean) {
        this.isVisible = isVisible
        val normalVisibility = if (isVisible) View.VISIBLE else View.INVISIBLE

		nowPlayingListHandle.findView().visibility = normalVisibility
        playerControlsLinearLayout.findView().visibility = normalVisibility
		ratingBarLinearLayout.findView().visibility = normalVisibility

		// Make this view gone so that song title text can take up full view when not displayed
        menuControlsLinearLayout.findView().visibility = if (isVisible) View.VISIBLE else View.GONE
    }
}
